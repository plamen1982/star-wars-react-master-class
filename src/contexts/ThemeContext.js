import { createContext } from 'react';
import { lightTheme } from '../styles';
const toggleTheme = () => {};

const ThemeContext = createContext({
  theme: { name: 'light', colors: {} },
  toggleTheme,
});

export default ThemeContext;
