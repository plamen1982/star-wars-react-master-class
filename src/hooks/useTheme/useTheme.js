import { useState, useEffect, useContext } from 'react';
import { darkTheme, lightTheme } from '../../styles';

function useTheme(ThemeContext) {
  const theme = useContext(ThemeContext);
  const [currentTheme, setCurrentTheme] = useState(theme);
  const toggleTheme = () => {
    setCurrentTheme(oldTheme => {
      if (oldTheme.name === 'light') {
        return {
          name: 'dark',
          colors: { ...darkTheme },
        };
      } else {
        return {
          name: 'light',
          colors: { ...lightTheme },
        };
      }
    });
  };

  useEffect(() => {
    toggleTheme();
  }, [theme]);

  return { currentTheme, toggleTheme };
}

export default useTheme;
