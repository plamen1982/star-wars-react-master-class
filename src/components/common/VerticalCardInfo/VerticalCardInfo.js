import React, { useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  Card,
  CardContent,
  Typography,
  Grid,
  CardMedia,
  Box,
} from '@material-ui/core';
import { ThemeContext } from '../../../contexts';

export default function VerticalCardInfo({ data, rowsToRender }) {
  const {
    currentTheme: {
      colors: { cards, defaultColors, links },
    },
  } = useContext(ThemeContext);
  const styleWithTheme = { cards, defaultColors, links };
  const useStyles = makeStyles(styleWithTheme);
  const classes = useStyles();
  return (
    <Grid item md={6} sm={12} xs={12}>
      <Box p={1}>
        <Card className={`${classes.defaultColors} ${classes.cards}`} p={3}>
          <Typography>
            <Box
              m={2}
              fontSize={25}
              fontFamily="Roboto"
              fontWeight="bold"
              className={classes.links}
            >
              {data.name}
            </Box>
          </Typography>
          <CardMedia
            className={classes.imageStyles}
            component="img"
            image={data.image}
            alt="url"
          />
          <CardContent>
            {rowsToRender.map(rowToRender => (
              <Typography
                className={classes.title}
                color={classes.cards.color}
                gutterBottom
              >
                {rowToRender.name}
                <Box
                  p={1}
                  component="span"
                  fontFamily="Roboto"
                  className={classes.links}
                >
                  {rowToRender.value}
                </Box>
              </Typography>
            ))}
          </CardContent>
        </Card>
      </Box>
    </Grid>
  );
}
