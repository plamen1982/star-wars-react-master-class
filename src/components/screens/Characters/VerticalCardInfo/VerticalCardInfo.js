import React, { useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {
  Card,
  CardContent,
  Typography,
  Grid,
  CardMedia,
  Box,
} from '@material-ui/core';
// import { styles } from './styles';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../../../../contexts';

export default function VerticalCardInfo({ data: character }) {
  const {
    currentTheme: {
      colors: { cards, defaultColors, links },
    },
  } = useContext(ThemeContext);
  const styleWithTheme = { cards, defaultColors, links };
  const useStyles = makeStyles(styleWithTheme);
  const classes = useStyles();
  return (
    <Grid item md={8} sm={12} xs={12}>
      <Box p={3}>
        <Card className={`${classes.defaultColors} ${classes.cards}`} p={3}>
          <Typography>
            <Box
              m={2}
              fontSize={25}
              fontFamily="Roboto"
              fontWeight="bold"
              className={classes.links}
            >
              {character.name}
            </Box>
          </Typography>
          <CardMedia
            className={classes.imageStyles}
            component="img"
            image={character.image}
            alt="url"
          />
          <CardContent>
            <Typography
              className={classes.title}
              color={classes.cards.color}
              gutterBottom
            >
              Height:
              <Box
                p={1}
                component="span"
                fontFamily="Roboto"
                className={classes.links}
              >
                {character.height}
              </Box>
            </Typography>
            <Typography
              className={classes.title}
              color={classes.cards.color}
              gutterBottom
            >
              Weight:
              <Box
                p={1}
                component="span"
                fontFamily="Roboto"
                className={classes.links}
              >
                {character.mass}
              </Box>
            </Typography>
            <Typography
              className={classes.title}
              color={classes.cards.color}
              gutterBottom
            >
              Species:
              <Box
                p={1}
                component="span"
                fontFamily="Roboto"
                className={classes.links}
              >
                {character.species.name}
              </Box>
            </Typography>
            <Typography
              className={classes.title}
              color={classes.cards.color}
              gutterBottom
            >
              Home World:
              <Box
                p={1}
                component="span"
                fontFamily="Roboto"
                className={classes.links}
              >
                {character.homeworld.name}
              </Box>
            </Typography>
          </CardContent>
        </Card>
      </Box>
    </Grid>
  );
}
