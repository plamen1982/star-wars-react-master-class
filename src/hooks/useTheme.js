import React, { useState, useEffect, useContext } from 'react';
import ThemeContext from '../contexts/ThemeContext';
import { darkTheme, lightTheme, commonColors } from '../styles';
const useTheme = () => {
  const theme = useContext(ThemeContext);
  const [currentTheme, setCurrentTheme] = useState(theme);
  const toggleTheme = () => {
    setCurrentTheme(oldTheme =>
      oldTheme.theme.name === 'light'
        ? {
            theme: { name: 'dark', colors: { ...darkTheme } },
          }
        : {
            theme: { name: 'light', colors: { ...lightTheme } },
          },
    );
  };

  useEffect(
    props => {
      toggleTheme();
    },
    [theme],
  );

  return { currentTheme, toggleTheme };
};

export default useTheme;
