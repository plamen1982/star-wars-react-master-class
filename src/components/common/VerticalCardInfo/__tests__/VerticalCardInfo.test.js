// import React from 'react';
// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import { makeStyles } from '@material-ui/core/styles';
// import { mount } from 'enzyme';
// import { ThemeContext } from '../../../../contexts';
// import VerticalCardInfo from '../VerticalCardInfo';
// import { lightTheme, darkTheme } from '../../../../styles';
// configure({ adapter: new Adapter() });

// describe('<VerticalCardInfo>', () => {
//   beforeEach(() => {});
//   test('should render correctly in debug mode', () => {
//     const toggleTheme = jest.fn();
//     const person = {
//       name: 'Luke Skywalker',
//       height: 172,
//       image:
//         'https://links.gunaxin.com/content/images/post_images/Luke_Skywalker_039_s_Original_039_Star_Wars_039_Lightsaber_Is_Going_Up_For_Auction_1544116268_4528.jpg',
//     };
//     const component = mount(
//       <ThemeContext.Provider value={{ theme, toggleTheme }}>
//         <VerticalCardInfo
//           data={person}
//           rowsToRender={[
//             { name: 'Height', value: 189 },
//             { name: 'Weight', value: 99 },
//             { name: 'Species', value: 'MockedName' },
//             { name: 'Home World', value: 'Hello World' },
//           ]}
//         />
//       </ThemeContext.Provider>,
//     );
//     expect(component).toMatchInlineSnapshot();
//   });
// });

// import { render, fireEvent } from '../../../../test-utils';
/* eslint global-require: 0 */
import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import { ThemeContext } from '../../../../contexts/index';
import VerticalCardInfo from '../VerticalCardInfo';

//TODO cannot import useTheme
import useTheme from '../../../../hooks/useTheme/useTheme';

describe('<VerticalCardInfo />', () => {
  let light;
  let wrapper;

  let theme;
  let renderHook;
  const ThemeContext = jest.fn();
  const useContextSpy = jest.spyOn(React, 'useContext');
  useContextSpy.mockImplementation(ThemeContext);
  beforeEach(() => {
    theme = require('../../../../styles');

    renderHook = () => {
      const HookWrapper = () => {
        return null;
      };

      return mount(<HookWrapper />);
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders without crashing', () => {
    wrapper = renderHook(
      <ThemeContext.Provider value={theme.darkTheme}>
        <VerticalCardInfo
          data={[]}
          title="Mocked title"
          image="mocked-image.jpeg"
          rowsToRender={[
            {
              name: 'mocked-name',
              value: 'mocked-value',
            },
          ]}
        />
      </ThemeContext.Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  console.log(wrapper);
});
