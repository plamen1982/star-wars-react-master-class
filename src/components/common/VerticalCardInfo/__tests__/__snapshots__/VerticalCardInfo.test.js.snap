// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalCardInfo /> renders without crashing 1`] = `<HookWrapper />`;
